File: app/models/ware.py
from sqlalchemy import Column, String, BigInteger, Numeric, DateTime
from sqlalchemy import UniqueConstraint, Index
from app.core.database import Base


class Ware(Base):
    """倉庫マスタ（WARE）- 全51カラム定義"""

    __tablename__ = "ware"

    # プライマリキー
    prkey = Column(BigInteger, primary_key=True, comment="")

    # 基本情報
    fctcd = Column(String(50), default="", comment="")
    whcd = Column(String(50), default="", comment="")
    whnm = Column(String(150), default="", comment="")

    # システム情報
    deldt = Column(DateTime, comment="")
    ludate = Column(DateTime, comment="")
    uuser = Column(String(20), default="", comment="")
    udate = Column(DateTime, comment="")

    # 表示情報
    dispno = Column(Numeric, default=0, comment="")
    whinfnm = Column(String(150), default="", comment="")

    # 住所・連絡先
    zip = Column(String(20), default="", comment="")
    add1 = Column(String(100), default="", comment="")
    add2 = Column(String(100), default="", comment="")
    email = Column(String(100), default="", comment="")
    tel = Column(String(20), default="", comment="")
    fax = Column(String(20), default="", comment="")
    whpicnm = Column(String(50), default="", comment="")

    # テンプレート・表示カウント
    trsrctempfile = Column(String(50), default="", comment="")
    trsrcdispcnt = Column(Numeric, default=0, comment="")
    trdesttempfile = Column(String(50), default="", comment="")
    trdestdispcnt = Column(Numeric, default=0, comment="")

    # グループ・タイプ
    whgrplcd = Column(String(50), default="", comment="")
    whgrpmcd = Column(String(50), default="", comment="")
    whtyp = Column(String(2), default="0", comment="")
    weicap = Column(Numeric, default=0, comment="")
    ocptyp = Column(String(2), default="", comment="")

    # フラグ
    isrecv = Column(String(2), default="", comment="")
    iscomp = Column(String(2), default="", comment="")
    iscalc = Column(String(2), default="", comment="")

    # 部門・印刷名
    deptcd = Column(String(50), default="", comment="")
    whprntnm = Column(String(150), default="", comment="")
    whprntnm2 = Column(String(150), default="", comment="")

    # 予約・在庫タイプ
    isrsv = Column(String(2), default="", comment="")
    isthrown = Column(String(2), default="", comment="")
    isvalue = Column(String(2), default="1", comment="")
    inorouttyp = Column(String(2), default="0", comment="")
    rsvstocgrpcd = Column(String(50), default="", comment="")

    # 受注グループ
    jobordgrp1 = Column(String(50), default="", comment="")
    disusewh = Column(String(1), default="0", comment="")

    # ブロック・配置情報
    blocknm = Column(String(50), default="", comment="")
    colnum = Column(Numeric, default=0, comment="")
    depthnum = Column(Numeric, default=0, comment="")
    stepnum = Column(Numeric, default=0, comment="")

    # 棚卸・ライン
    shemanatyp = Column(String(2), default="", comment="")
    linecd = Column(String(50), default="", comment="")

    # 追加情報1-5
    wareaddinfo01 = Column(String(50), default="", comment="")
    wareaddinfo02 = Column(String(50), default="", comment="")
    wareaddinfo03 = Column(String(50), default="", comment="")
    wareaddinfo04 = Column(String(50), default="", comment="")
    wareaddinfo05 = Column(String(50), default="", comment="")

    # 前倉庫
    prewhcd = Column(String(50), default="", comment="")

    # 制約定義
    __table_args__ = (
        UniqueConstraint("fctcd", "whcd", "deldt", name="ware_fctcd_whcd_deldt"),
        UniqueConstraint("fctcd", "whcd", name="ware_fctcd_whcd_key"),
        Index("ware_pkey", "prkey", unique=True),
    )

    # ========================================
    # リレーション定義
    # ========================================
    # 今後必要に応じてリレーションを追加

    def __repr__(self):
        return f"<Ware(prkey={self.prkey}, fctcd={self.fctcd}, whcd={self.whcd}, whnm={self.whnm})>"

File: src/BaggingInstructions.Api/Entities/Ware.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace BaggingInstructions.Api.Entities.Legacy;

[Table("ware")]
public class Ware
{
    [Column("prkey")]
    public long Prkey { get; set; }

    [Column("fctcd")]
    public string? Fctcd { get; set; } = "";

    [Column("whcd")]
    public string? Whcd { get; set; } = "";

    [Column("whnm")]
    public string? Whnm { get; set; } = "";

    [Column("deldt")]
    public DateTime? Deldt { get; set; }

    [Column("ludate")]
    public DateTime? Ludate { get; set; }

    [Column("uuser")]
    public string? Uuser { get; set; } = "";

    [Column("udate")]
    public DateTime? Udate { get; set; }

    [Column("dispno")]
    public decimal Dispno { get; set; }

    [Column("whinfnm")]
    public string? Whinfnm { get; set; } = "";

    [Column("zip")]
    public string? Zip { get; set; } = "";

    [Column("add1")]
    public string? Add1 { get; set; } = "";

    [Column("add2")]
    public string? Add2 { get; set; } = "";

    [Column("email")]
    public string? Email { get; set; } = "";

    [Column("tel")]
    public string? Tel { get; set; } = "";

    [Column("fax")]
    public string? Fax { get; set; } = "";

    [Column("whpicnm")]
    public string? Whpicnm { get; set; } = "";

    [Column("trsrctempfile")]
    public string? Trsrctempfile { get; set; } = "";

    [Column("trsrcdispcnt")]
    public decimal Trsrcdispcnt { get; set; }

    [Column("trdesttempfile")]
    public string? Trdesttempfile { get; set; } = "";

    [Column("trdestdispcnt")]
    public decimal Trdestdispcnt { get; set; }

    [Column("whgrplcd")]
    public string? Whgrplcd { get; set; } = "";

    [Column("whgrpmcd")]
    public string? Whgrpmcd { get; set; } = "";

    [Column("whtyp")]
    public string? Whtyp { get; set; } = "0";

    [Column("weicap")]
    public decimal Weicap { get; set; }

    [Column("ocptyp")]
    public string? Ocptyp { get; set; } = "";

    [Column("isrecv")]
    public string? Isrecv { get; set; } = "";

    [Column("iscomp")]
    public string? Iscomp { get; set; } = "";

    [Column("iscalc")]
    public string? Iscalc { get; set; } = "";

    [Column("deptcd")]
    public string? Deptcd { get; set; } = "";

    [Column("whprntnm")]
    public string? Whprntnm { get; set; } = "";

    [Column("whprntnm2")]
    public string? Whprntnm2 { get; set; } = "";

    [Column("isrsv")]
    public string? Isrsv { get; set; } = "";

    [Column("isthrown")]
    public string? Isthrown { get; set; } = "";

    [Column("isvalue")]
    public string? Isvalue { get; set; } = "1";

    [Column("inorouttyp")]
    public string? Inorouttyp { get; set; } = "0";

    [Column("rsvstocgrpcd")]
    public string? Rsvstocgrpcd { get; set; } = "";

    [Column("jobordgrp1")]
    public string? Jobordgrp1 { get; set; } = "";

    [Column("disusewh")]
    public string? Disusewh { get; set; } = "0";

    [Column("blocknm")]
    public string? Blocknm { get; set; } = "";

    [Column("colnum")]
    public decimal Colnum { get; set; }

    [Column("depthnum")]
    public decimal Depthnum { get; set; }

    [Column("stepnum")]
    public decimal Stepnum { get; set; }

    [Column("shemanatyp")]
    public string? Shemanatyp { get; set; } = "";

    [Column("linecd")]
    public string? Linecd { get; set; } = "";

    [Column("wareaddinfo01")]
    public string? Wareaddinfo01 { get; set; } = "";

    [Column("wareaddinfo02")]
    public string? Wareaddinfo02 { get; set; } = "";

    [Column("wareaddinfo03")]
    public string? Wareaddinfo03 { get; set; } = "";

    [Column("wareaddinfo04")]
    public string? Wareaddinfo04 { get; set; } = "";

    [Column("wareaddinfo05")]
    public string? Wareaddinfo05 { get; set; } = "";

    [Column("prewhcd")]
    public string? Prewhcd { get; set; } = "";
}
